libraries/abstractions/common_io/include/iot_spi.h

```diff
@@ -244,94 +298,117 @@ int32_t iot_spi_write_sync( IotSPIHandle_t const pxSPIPeripheral,
  * @brief The SPI master starts transmission of data to the slave
  * asynchronously.
  *
- * The application should call this function to start asynchronous write.
- * The application should also give the address of the buffer to send the data
- * from; with number of bytes to write.
+ * This function attempts to read certain number of bytes from a pre-allocated buffer, in asynchronous way.
+ * It returns immediately when the operation is started and the status can be check by calling iot_spi_ioctl.
+ * Once the operation completes, successful or not, the user callback will be invoked.
+ *
+ * Partial read might happen, e.g. slave device unable to receive more data.
+ * And the number of bytes that have been actually read can be obtained by calling iot_spi_ioctl.
```

Comment:
Similar here. please clarify that Callback will NOT be called on partial writes unless there is a success or error.

Reply:
And the number of bytes that have been actually **read.** -> Should be write?

Reply:
Yes, fixed it